ns1/data_source_record.go

```diff
@@ -0,0 +1,118 @@
+package ns1
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	ns1 "gopkg.in/ns1/ns1-go.v2/rest"
+)
+
+func dataSourceRecord() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ttl": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"meta": {
+				Type:             schema.TypeMap,
+				Optional:         true,
+				DiffSuppressFunc: metaDiffSuppressUp,
```

Comment:
`Optional` and `DiffSupressFunc` are unnecessary on data source attributes AFAIK since these fields have no configuration.  Similarly I believe these should all contain `Computed: true` as they are not user configurable.